sql/core/src/main/scala/org/apache/spark/sql/internal/BaseSessionStateBuilder.scala

```diff
@@ -181,9 +181,14 @@ abstract class BaseSessionStateBuilder(
    * Note: this depends on the `conf` and `catalog` fields.
    */
   protected def analyzer: Analyzer = new Analyzer(catalogManager) {
+
+    override val extendedSubstitutionRules: Seq[Rule[LogicalPlan]] =
+      Seq(ViewSubstitution(sqlParser))
```

Comment:
why does it need to be a substitution rule? The v1 view is resolved in the main resolution batch.

Reply:
ViewSubstitution had to be done in the same batch as CTESubstitution to handle nested cte or view in each other.

However, since there have been many changes in master since I originally developed the code, this requirement may no longer be necessary anymore. Let me revisit. It'd be much better to move the rule to Resolution batch if possible!